processing/src/main/java/io/druid/query/expression/ExprUtils.java

```diff
@@ -73,7 +74,14 @@ public static PeriodGranularity toPeriodGranularity(
     } else {
       Chronology chronology = timeZone == null ? ISOChronology.getInstanceUTC() : ISOChronology.getInstance(timeZone);
       final Object value = originArg.eval(bindings).value();
-      origin = value != null ? new DateTime(value, chronology) : null;
+      if (value instanceof String && StringUtils.isEmpty((String) value)) {
```

Comment:
This comment: https://github.com/apache/incubator-druid/pull/5452#discussion_r200339008 is not addressed

Reply:
https://github.com/apache/incubator-druid/issues/6040
Will fix this line in this PR and prohibit the api in future PR.

Reply:
done.